tests/syntax_tests/data/printer/typexpr/expected/arrow.res.txt

```diff
@@ -132,7 +132,7 @@ type t = @attr (foo, @attr2 ~f: bar, @attr3 ~f: baz) => unit
 type t = @attr (string => @attr (int => unit))
 type t = @attr (string, int) => @attr (int, float) => unit
 type t = @attr (int => @attr (int, float) => @attr (unit => unit => unit))
-type t = @attr (@attr2 ~f: int, @attr3 ~g: float) => unit
+type t = (@attr @attr2 ~f: int, @attr3 ~g: float) => unit
```

Comment:
This change doesn't look correct?

Reply:
There's a single node.
No way to distinguish inside and outside the parens.

Reply:
Later on, a good representation should probably take a list of params, each one with its own locations and attributes.
Then the single function node could have its own location and attributes.

We're bound to having to convert back and forth with the original parse tree for ast compatibility, so either one would have to preserve all information,or at least all the info required for the later phases of the compiler (other than pretty printing).

Reply:
> I thought `@attr` was on the whole function and `@attr2` was on the first arg.
> 
> So what exactly are `@attr` and `@attr2` on now?

The only attributes are at the level of ast nodes.
The type (a, b) => c is just a pretty printed form for 2 ast nodes. (The outer one has the arity info).

Reply:
So (a,b) => c is really a =2=> (b => c)
Attributes are on the arrow nodes so the  there's no distinction between an attribute on a, and on the whole type: it's the same node, the outer one.

Reply:
Cleaning up these things is exactly what we need custom asts for.
This aspect can be cleaned up in a further refinement.